### PR TITLE
Add wrappers that transforms PointCollectors in to ContinuousCollectors

### PR DESCRIPTION
**Pull request by haq204**
## Summary
- ContinuousPointCollector wraps a PointCollector into a ContinuousCollector that calls Collect() on an interval

- OnceContinuousCollector is a ContinuousCollector that wraps a PointCollector and performs a one-shot collection.

  Start() will call Collect() once and return the result in a channel with a buffer size of 1 and then close it. This is useful for   collectors that only need to run once because the information collected doesn't change (e.g. hardware info).

## Test plan
- [x] Run existing collector tests to ensure no regressions
- [x] New tests cover edge cases and error conditions
- [x] Verify channel lifecycle management works correctly

### CODE REVIEW

File: pkg/performance/registry.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		return fmt.Errorf("continuous collector for metric type %s already registered", metricType)
+	}
+	if _, exists := r.pointCollectors[metricType]; exists {
+		return fmt.Errorf("point collector for metric type %s already registered", metricType)
+	}
+
+	r.continuousCollectors[metricType] = collector
+	r.logger.Info("registered continuous collector", "type", metricType, "name", collector.Name())
+	return nil
+}
```
Comment (jra3) [inline pkg/performance/registry.go:63]: Hrmm... I think we are missing that we might have different collector instances? E.g. We might have some collector for each container on a machine, but they would have the same metricType.

Comment (haq204) [reply pkg/performance/registry.go:63]: I just moved the registry stuff to a separate file without changes. Going to follow up on refactoring the registry in a separate PR!
